core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -1048,53 +951,40 @@ snoc = snocWith f
 -- Random reads
 -------------------------------------------------------------------------------
 
+-- XXX Can this be deduplicated with array/foreign
```

Comment:
What does this mean?

Reply:
Data.Array.Foreign also has getIndex/getIndexSafe functions with similar code. Maybe we can use common implementation here, passing a liftio argument.